src/librustdoc/html/render.rs

```diff
@@ -210,6 +206,25 @@ crate struct SharedContext {
     pub generate_redirect_pages: bool,
     /// The fs handle we are working with.
     pub fs: DocFS,
+    /// The default edition used to parse doctests.
+    pub edition: Edition,
+    pub codes: ErrorCodes,
+    playground: Option<markdown::Playground>,
+}
+
+impl Context {
+    fn path(&self, filename: &str) -> PathBuf {
+        let mut iter = filename.splitn(2, '.');
```

Comment:
I'm not a big fan of this: why not converting into a path and then get base and extension from it?

Reply:
I think in this case it'd be equivalent (I can switch to that) but when I wrote this I was thinking we might have something like `style.min.css` at some point, and the extension method on paths only splits at the last `.` so we'd get `style.min-suffix.css` instead of `style-suffix.min.css`.

Reply:
I see. In this case this is indeed better. Can you add a comment giving this explanation so people don't change it by mistake please?

Reply:
Added the comment.